app/utils/Validation.scala

```diff
@@ -0,0 +1,22 @@
+package utils
+
+import org.joda.time.YearMonth
+import org.joda.time.format.DateTimeFormat
+
+import scala.util.Try
+
+/**
+ * Created by coolit on 16/03/2018.
+ */
+object Validation {
+  private val minKeyLength: Int = 5
+  private val maxKeyLength: Int = 20
+  private val periodFormat: String = "yyyyMM"
+  private val validCategories: List[String] = List("ENT", "LEU", "VAT", "PAYE", "CH")
```

Comment:
Set[String] would be a conceptually better representation if you are only interested in membership (and also offers much better performance).